services/worker/src/worker/job_runners/dataset/is_valid.py

```diff
@@ -26,22 +30,54 @@ def compute_is_valid_response(dataset: str, processing_graph: ProcessingGraph) -
         dataset (`str`):
             A namespace (user or an organization) and a repo name separated
             by a `/`.
-        processing_graph (`ProcessingGraph`):
-            The processing graph. In particular, it must provide the list of
-            processing steps that enable the viewer and the preview.
     Returns:
-        `DatasetIsValidResponse`: The response (viewer, preview).
+        `Tuple[IsValidResponse, float]`: The response (viewer, preview, search) and the progress.
     """
-    logging.info(f"get is-valid response for dataset={dataset}")
+    logging.info(f"get is-valid response for {dataset=}")
 
-    viewer = is_valid_for_kinds(
-        dataset=dataset, kinds=[step.cache_kind for step in processing_graph.get_processing_steps_enables_viewer()]
-    )
-    preview = is_valid_for_kinds(
-        dataset=dataset, kinds=[step.cache_kind for step in processing_graph.get_processing_steps_enables_preview()]
-    )
+    config_names_response = get_previous_step_or_raise(kinds=["dataset-config-names"], dataset=dataset)
+    content = config_names_response.response["content"]
+    if "config_names" not in content:
+        raise PreviousStepFormatError("Previous step did not return the expected content: 'config_names'.")
+
+    preview_count = 0
+    viewer_count = 0
+    search_count = 0
+    try:
+        total = 0
+        pending = 0
+        for config_item in content["config_names"]:
+            config = config_item["config"]
+            total += 1
+            try:
+                response = get_response(kind="config-is-valid", dataset=dataset, config=config)
+            except CacheEntryDoesNotExistError:
+                logging.debug("No response found in previous step for this dataset: 'config-is-valid'.")
+                pending += 1
+                continue
+            if response["http_status"] != HTTPStatus.OK:
+                logging.debug(f"Previous step gave an error: {response['http_status']}.")
+                continue
+            split_is_valid_content = response["content"]
+            preview_count += 1 if split_is_valid_content["preview"] else 0
+            viewer_count += 1 if split_is_valid_content["viewer"] else 0
+            search_count += 1 if split_is_valid_content["search"] else 0
+    except Exception as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.", e) from e
 
-    return DatasetIsValidResponse({"viewer": viewer, "preview": preview})
+    progress = (total - pending) / total if total else 1.0
+    preview = preview_count == total
```

Comment:
same comment: should be or, not and, and it might be clearer to use logical operators instead of a sum

Reply:
Makes more sense, done in https://github.com/huggingface/datasets-server/pull/1618/commits/6f0bda0f19285519098a66d5eb90b82cc545f572